### PR TITLE
mhaikalr calendar updates

### PR DESCRIPTION
**Pull request by mhaikalr**

### CODE REVIEW

File: src/lifetracker/calendar/CalendarListImpl.java [hunk 1]
```diff
@@ -29,19 +31,21 @@
...
      *
-     * @see lifetracker.calendar.CalenderList#add(java.lang.String,
-     * java.time.LocalDateTime)
+     * @see lifetracker.calendar.CalenderList#add(java.lang.String)
      */
     @Override
     public void add(String name) {
-        CalendarEntryImpl ft = new CalendarEntryImpl(name, null, null);
-        taskList.add(ft);
+        int idToSet = taskList.lastKey() + 1;
```
Comment (yichenshen) [inline src/lifetracker/calendar/CalendarListImpl.java]: The id should be unique across both lists. You need to check both Maps for the last key.

File: src/lifetracker/calendar/CalendarListImpl.java [hunk 2]
```diff
@@ -52,8 +56,9 @@ public void add(String name) {
      */
     @Override
     public void add(String name, LocalDateTime deadline) {
-        CalendarEntryImpl dt = new CalendarEntryImpl(name, null, deadline);
-        taskList.add(dt);
+        int idToSet = taskList.lastKey() + 1;
```
Comment (yichenshen) [inline src/lifetracker/calendar/CalendarListImpl.java]: Same here

File: src/lifetracker/calendar/CalendarListImpl.java [hunk 3]
```diff
@@ -64,8 +69,39 @@ public void add(String name, LocalDateTime deadline) {
      */
     @Override
     public void add(String name, LocalDateTime start, LocalDateTime end) {
-        CalendarEntryImpl e = new CalendarEntryImpl(name, start, end);
-        eventList.add(e);
+        int idToSet = eventList.lastKey() + 1;
```
Comment (yichenshen) [inline src/lifetracker/calendar/CalendarListImpl.java]: And here.

**Comment by mhaikalr** [comment]
@yichenshen issues you mentioned addressed. Pushed to the same branch, see latest commit.

**Pull request merged**